server/helpers/filters.js

```diff
@@ -1,38 +1,31 @@
-export const filterStringKeys = (filterQuery, stringArray, query, prefix = '') => {
-  let pre = '';
-  if (prefix.length > 0) {
-    pre = `${prefix}.`;
-  }
-
-  stringArray.forEach((key) => {
-    if (query[key]) {
-      filterQuery.push({ [`${pre}${key}`]: { $regex: query[key], $options: 'i' } });
-    }
-  });
+export const FILTER_TYPE = {
+  BOOLEAN: 'boolean',
+  INTEGER: 'integer',
+  STRING: 'string',
 };
 
-export const filterIntegerKeys = (filterQuery, integerArray, query, prefix = '') => {
-  let pre = '';
-  if (prefix.length > 0) {
-    pre = `${prefix}.`;
-  }
+export const buildFilterQuery = (arrayWithKeys, query, type, prefix = '') => {
```

Comment:
Lets use `filterKeys` here, `arrayWithKeys` is not descriptive.

```suggestion
export const buildFilterQuery = (filterKeys, query, type, prefix = '') => {
```